patch based on Alan Cox suggestion. full thread available at:
http://www.spinics.net/lists/linux-ide/msg25854.html

--- linux-2.6.25/drivers/ata/ahci.c.orig	2008-09-04 13:30:27.000000000 +0200
+++ linux-2.6.25/drivers/ata/ahci.c	2008-09-04 18:51:14.000000000 +0200
@@ -590,10 +590,6 @@
 	{ PCI_VDEVICE(SI, 0x1185), board_ahci }, /* SiS 966 */
 	{ PCI_VDEVICE(SI, 0x0186), board_ahci }, /* SiS 968 */
 
-	/* Marvell */
-	{ PCI_VDEVICE(MARVELL, 0x6145), board_ahci_mv },	/* 6145 */
-	{ PCI_VDEVICE(MARVELL, 0x6121), board_ahci_mv },	/* 6121 */
-
 	/* Generic, PCI class code for AHCI */
 	{ PCI_ANY_ID, PCI_ANY_ID, PCI_ANY_ID, PCI_ANY_ID,
 	  PCI_CLASS_STORAGE_SATA_AHCI, 0xffffff, board_ahci },
